concurrency/SynchronizationContext.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "concurrency/MpmcQueue.h"
+#include "utils/Function.h"
+
+namespace concurrency
+{
+
+class SynchronizationContext
```

Comment:
This holds a reference to a queue but is no longer a queue. The goal is provide a way to add Functions to the queue without make available other methods like `pop`.

@olofkallander, I didn't called a queue because is no longer a queue, is more a wrapper to a queue where the push methods are available. I based the name on `SynchronizationContext` available on .net framework, please suggest a new name if you think this is not a good one